app/scripts/controllers/permissions/index.js

```diff
@@ -424,16 +424,18 @@ export class PermissionsController {
 
       // caveat names are unique, and we will only construct this caveat here
       ethAccounts.caveats = ethAccounts.caveats.filter((c) => (
-        c.name !== CAVEAT_NAMES.exposedAccounts
+        c.name !== CAVEAT_NAMES.exposedAccounts && c.name !== CAVEAT_NAMES.primaryAccountOnly
       ))
 
+      ethAccounts.caveats.push({ type: 'limitResponseLength', value: 1, name: CAVEAT_NAMES.primaryAccountOnly })
       ethAccounts.caveats.push(
         {
           type: 'filterResponse',
           value: finalizedAccounts,
           name: CAVEAT_NAMES.exposedAccounts,
         },
       )
+
```

Comment:
extreme nit
```suggestion
```

Reply:
i made some changes since this suggestion and I'm not sure if it accurately represents your original concern. If it does I'll happily remove the line!

Reply:
Haha, my original concern was just an unnecessary newline, which is no longer there!